### PR TITLE
HV-1531 Regroup the ValidatorFactory scoped properties in ValidatorFactoryContext

### PR DESCRIPTION
**Pull request by marko-bekhta**
- https://hibernate.atlassian.net/browse/HV-1531

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java [hunk 1]
```diff
@@ -75,36 +76,9 @@
 	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
 
 	/**
-	 * The default message interpolator for this factory.
+	 * Context containing all {@link HibernateValidatorFactory} level helpers and configuration properties.
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: I think we can refer to the generic `ValidatorFactory` here. It's more to compare with `Validator` or validation-scoped elements.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java [hunk 2]
```diff
@@ -678,4 +625,202 @@ public String toString() {
...
+					clockProvider,
+					temporalValidationTolerance,
+					scriptEvaluatorFactory,
+					failFast,
+					traversableResolverResultCacheEnabled
+			);
+		}
+
+
+
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: Too many new lines.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java [hunk 3]
```diff
@@ -678,4 +625,202 @@ public String toString() {
...
+				return this;
+			}
+
+			public Builder setTraversableResolverResultCacheEnabled(boolean traversableResolverResultCacheEnabled) {
+				this.traversableResolverResultCacheEnabled = traversableResolverResultCacheEnabled;
+				return this;
+			}
+
+			public ValidatorFactoryContext build() {
+				return new ValidatorFactoryContext( messageInterpolator, traversableResolver, parameterNameProvider, clockProvider, temporalValidationTolerance, scriptEvaluatorFactory, failFast, traversableResolverResultCacheEnabled );
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: I would be surprised if it's all on a 120 characters line :).

A local ctrl+shift+f will help.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 4]
```diff
@@ -137,25 +133,18 @@
...
-			Duration temporalValidationTolerance) {
+			ValidatorFactoryContext validatorFactoryContext) {
 		this.constraintValidatorFactory = constraintValidatorFactory;
 		this.beanMetaDataManager = beanMetaDataManager;
 		this.valueExtractorManager = valueExtractorManager;
 		this.constraintValidatorManager = constraintValidatorManager;
 		this.validationOrderGenerator = validationOrderGenerator;
-		this.traversableResolver = traversableResolver;
-		this.validatorScopedContext = new ValidatorScopedContext( messageInterpolator, parameterNameProvider, clockProvider, temporalValidationTolerance, scriptEvaluatorFactory, failFast, traversableResolverResultCacheEnabled );
+		this.validatorScopedContext = validatorFactoryContext.toValidatorScopedContext();
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: I think I would prefer the opposite i.e. having a `ValidatorScopedContext` constructor taking a validator factory context. We might want to add other things in it at one point.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java [hunk 5]
```diff
@@ -678,4 +625,202 @@ public String toString() {
 					+ ", methodValidationConfiguration=" + methodValidationConfiguration + "]";
 		}
 	}
+
+	static class ValidatorFactoryContext {
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: Let's call it `ValidatorFactoryScopedContext` for symmetry.

Same for the builder.

Comment (marko-bekhta) [reply engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: do you mean:
`ValidatorFactoryContext` -> `ValidatorFactoryScopedContext`
`Builder` -> `ValidatorFactoryScopedContextBuilder` ? 
I followed the same naming for a builder as for example in `MethodValidationConfiguration.Builder`, or am I missing something :smiley:

Comment (gsmet) [reply engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java]: Ah yes, sorry, keep the builder as it is :).

**Comment by gsmet** [comment]
Jenkins, retest this please

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by gunnarmorling** [comment]
Nice, thanks @marko-bekhta!

**Pull request closed without merging**